src/cmd/sanity_test/sanity_tests.go

```diff
@@ -135,29 +133,33 @@ func checkEmailRestoration(
 	}
 
 	for _, fld := range childFolder.GetValue() {
-		var (
-			fldID   = ptr.Val(fld.GetId())
-			fldName = ptr.Val(fld.GetDisplayName())
-			count   = ptr.Val(fld.GetTotalItemCount())
-			ictx    = clues.Add(
-				ctx,
-				"child_folder_id", fldID,
-				"child_folder_name", fldName,
-				"expected_count", itemCount[fldName],
-				"actual_count", count)
-		)
+		restoreDisplayName := ptr.Val(fld.GetDisplayName())
 
-		if itemCount[fldName] != count {
-			logger.Ctx(ictx).Error("test failure: Restore item counts do not match")
+		// check if folder is the data folder we loaded or the base backup to verify
+		// the incremental backup worked fine
+		if strings.EqualFold(restoreDisplayName, dataFolder) || strings.EqualFold(restoreDisplayName, baseBackupFolder) {
+			count, _ := ptr.ValOK(fld.GetTotalItemCount())
+
+			restoreItemCount[restoreDisplayName] = count
+			checkAllSubFolder(ctx, client, fld, testUser, restoreDisplayName, dataFolder, restoreItemCount)
+		}
+	}
+
+	verifyEmailData(ctx, restoreItemCount, itemCount)
+}
+
+func verifyEmailData(ctx context.Context, restoreMessageCount, messageCount map[string]int32) {
+	for fldName, emailCount := range messageCount {
+		if restoreMessageCount[fldName] != emailCount {
+			logger.Ctx(ctx).Error("test failure: Restore item counts do not match")
```

Comment:
logs should contain the comparison values.
```suggestion
			logger.Ctx(ctx).Errorw(
			"test failure: Restore item counts do not match",
			"expected", emailCount,
			"actual", restoreMessageCount[fldName])
```